sdk/helper/testhelpers/schema/response_validation.go

```diff
@@ -66,6 +66,24 @@ func validateResponseDataImpl(schema *framework.Response, data map[string]interf
 		return fmt.Errorf("failed to unmashal data: %w", err)
 	}
 
+	// these are special fields that will not show up in the final response and
+	// should be ignored
+	for _, field := range []string{
+		logical.HTTPContentType,
+		logical.HTTPRawBody,
+		logical.HTTPStatusCode,
+		logical.HTTPRawBodyAlreadyJSONDecoded,
+		logical.HTTPCacheControlHeader,
+		logical.HTTPPragmaHeader,
+		logical.HTTPWWWAuthenticateHeader,
```

Comment:
maybe we should create a ticket to go though and update endpoints that still use these fields

Reply:
We use the first three a lot in PKI, and to my knowledge, this is unavoidable based on the protocols we implement.

Reply:
Do you mean update the responses that use this?

Reply:
There was only one that I removed here https://github.com/hashicorp/vault/pull/19530/commits/3c228e9fe1055b3e1cdd76b9af9ed26eed1125a1. This test will not pass with any endpoints that use these fields.

Reply:
I think I only checked in that one. I've already removed these fields from this [PR](https://github.com/hashicorp/vault/pull/18612) as well

Reply:
@cipherboy, to clarify, this test only concerns itself with the declared response schemas. If these fields are used in the actual responses, they will be simply ignored by the test for validity checking purposes :)

Reply:
Right, sorry. I thought @dhuckins was suggesting we remove the usage in the endpoint implementations themselves, which I was a little concerned about :-D All good here!